### DISCUSSION TITLE
Several targets for a form, some of them optional

### DISCUSSION THREAD

**Discussion by msurdi** [unpoly/unpoly#303] [Q&A] [answered]
Hi!

I have a question about ignoring targets in form submissions, hopefully somebody has seen this situation before and cal help me find the best way to do this.

I have a view such as:

```html
<form up-target="#section1, #section2, #section3" ...>...</form>
<div id="section1">...</div>
<div id="section2">...</div>
<div id="section3">...</div>
```

The response to the form submission, sometimes might include just one section, or two, and not all of them depending on business logic in the backend. This works, but I get an error in the console:

```
Caught (in promise) up.Failed: Could not find common target in current page and response (tried selectors [#section1, #section2, #section3] in root layer)....
```

Is there any way to indicate Unpoly that some of those targets might not exist in the response, but that it is OK and shouldn't complain about it? 

Thanks!

**Comment by triskweline** [comment, marked as answer]
All selectors in `[up-target]` must be present or an error is thrown.

Some alternatives:

- Target a container element that contains all the sections.
- Mark optional sections with `[up-hungry]` and don't include them in the `[up-target]` attribute.
- Always include all sections in the form, but render them with empty content when not needed. You can hide them via CSS with `#section:empty { display: none }`.

**Reply by msurdi** [reply]
I totally forgot about the up-hungry. Looks like it is the best solution to my problem. I love Unpoly.

Thanks!!!
